contracts/PoolManager.sol

```diff
@@ -0,0 +1,150 @@
+                }
+            }
+        } else {
+            for (uint i = 0; i < _pools.length; i++) { 
+                if (!fullPoolList.contains(_pools[i])) {
+                    fullPoolList.add(_pools[i]);
+                }
+            }
+        }
+
+        emit AddPools(_pools, _isLegacy);
+    }
+
+    function removePools(address[] calldata _pools, bool _isLegacy) public onlyAuthorized {
+        if (_isLegacy) {
+            for (uint i = 0; i < _pools.length; i++) { 
+                legacyPoolList.remove(_pools[i]);
+            }
+        } else {
+            for (uint i = 0; i < _pools.length; i++) { 
+                fullPoolList.add(_pools[i]);
+            }
+        }
+
+        emit RemovePools(_pools, _isLegacy);
+    }
+
+    function changeGovernanceToken(IERC20 _governanceToken) public onlyOwner {
+        governanceToken = _governanceToken;
+
+        emit GovernanceTokenChange(_governanceToken);
+    }
+
+    /*
+        Read Functions - Public 
+    */
+
+    function allPools() external view returns (address[] memory) {
+        return fullPoolList.values();
+    }
+
+    function allLegacyPools() external view returns (address[] memory) {
+        return legacyPoolList.values();
+    }
+
+    function viewTotalGovernanceHoldings(address userAddress) external view returns (uint256) {
+        uint256 stakedBalance;
+        uint256 holdingBalance = governanceToken.balanceOf(userAddress);
+        address[] memory activePools = this.allActivePools();
+        
+        for (uint256 i = 0; i < activePools.length; i++) { 
+            IRewardPool pool = IRewardPool(activePools[i]);
+            if (legacyPoolList.contains(activePools[i])) {
+                if (pool.stakeToken() == address(governanceToken)) {
+                    IRewardPool.UserInfo memory userInfo = pool.userInfo(userAddress);
+                    uint256 stakedAmount = userInfo.amount;
+                    stakedBalance += stakedAmount;
+                }
+            } else {
+                if (pool.STAKE_TOKEN() == address(governanceToken)) {
+                    IRewardPool.UserInfo memory userInfo = pool.userInfo(userAddress);
+                    uint256 stakedAmount = userInfo.amount;
+                    stakedBalance += stakedAmount;
+                }
+            }
+        }
+
+        return holdingBalance + stakedBalance;
+    }
+
+    function getActivePoolCount() external view returns (uint) {
+        uint256 count = 0;
+
+        for (uint256 i = 0; i < fullPoolList.length(); i++) { 
+            address pool = fullPoolList.at(i);
+            uint256 endBlock = IRewardPool(pool).bonusEndBlock();
+            if (endBlock > block.number) {
+                ++count;
+            }
+        }
+
+        return count;
+    }
+
+    function allActivePools() external view returns (address[] memory) {
+        address[] memory _activePoolList = new address[](this.getActivePoolCount());
```

Comment:
`this` keyword?

Reply:
Without the keyword, I seem to be getting this error 🤔
<img width="637" alt="image" src="https://user-images.githubusercontent.com/78942906/179425192-5ca1b03e-3061-497b-b758-88096cc4fd5e.png">